### PR TITLE
use OpenAstronomy build workflow

### LINKED ISSUE #39: BLD: Use OpenAstronomy publish workflow, provide a lot of wheels
Since you have C-extension, you need to generate a wheel for different distribution combos. The spacetelescope workflow is very limited. I propose you switch to OpenAstronomy instead.

https://github.com/OpenAstronomy/github-actions-workflows#build-and-publish-a-python-package

Example usage in production: https://github.com/astropy/astropy/blob/main/.github/workflows/publish.yml

### PR DESCRIPTION
**Pull request by zacharyburnett**
resolves #39

this PR changes the build workflow to use the OpenAstronomy one, and also adds a build-only (without publishing to PyPI) run to pull requests

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      targets: |
+        # Linux wheels
+        - cp3*-manylinux_x86_64
+        # MacOS wheels
+        - cp3*-macosx_x86_64
+        # Until we have arm64 runners, we can't automatically test arm64 wheels
+        - cp3*-macosx_arm64
+      sdist: true
+    secrets:
+      pypi_token: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER }}
```
Comment (pllim) [inline .github/workflows/build.yml:27]: I am not sure if you can reuse the old password in this workflow. Do you know for sure this will work?

**Review by pllim — COMMENTED** [review]
Why no windows?

**Comment by mcara** [comment]
This is the same workflow [used in `drizzle`](https://github.com/spacetelescope/drizzle/blob/master/.github/workflows/build.yml) which, unfortunately, [does not create source distribution](https://pypi.org/project/drizzle/#files) and also does not create Windows wheels.

**Comment by pllim** [comment]
Source distribution should be pretty standard. Windows support is negotiable, though for something that only does histogram calculations, I don't see why Windows support should be excluded; not like you pulling in crds or jwst over here.

**Review by pllim — APPROVED** [review]
I pushed a commit to add Windows wheels.

Not sure about the PyPI password but I  guess you will find out when you try to release next time.

File: .github/workflows/build.yml (hunk 1 above)
Comment (zacharyburnett) [reply .github/workflows/build.yml:30]: yes indeed, the `PYPI_PASSWORD_STSCI_MAINTAINER` secret is actually an API token for the PyPI account (I dunno why they named it that)

**Pull request merged**